src/query/procedure/module.cpp

```diff
@@ -133,6 +135,20 @@ void RegisterMgLoad(ModuleRegistry *module_registry, utils::RWLock *lock, Builti
   module->AddProcedure("load", std::move(load));
 }
 
+namespace {
+[[nodiscard]] bool IsFileEditable(const std::optional<std::filesystem::path> &path) {
+  return path && access(path->c_str(), W_OK) == 0;
+}
```

Comment:
I am just wondering from the users perspective what does this mean?
Since if the file is editable it might not be executable nor readable, and therefore unusable if it cannot be expected as a query module?

Reply:
If something of that is true, then it wouldn't be loaded as a module.
This assumes all the libs are valid alread.y